src/esi/Element.h

```diff
@@ -83,5 +85,13 @@ class ESIElement : public esiTreeParent
     virtual void finish() = 0;
 };
 
+/// Call finish() and set to nil the given element. Element may already be nil.
+/// When element is part of a set, use pos to indicate position/ID
+/// for debugging.
+extern void FinishAnElement(ESIElement::Pointer &, int pos = -1);
```

Comment:
I am leaving it out of scope for several reasons; it is just cleanup unrelated to GCC needs, involves a lot of line re-writes just for the name change, and cleanup moving these functions callers into namespace Esi later would involve re-touching all those lines to remove the new Esi:: prefix.

Reply:
> it is just cleanup unrelated to GCC needs

You are replacing old functions for "GCC needs" reasons (presumably). Selecting the best new function name is certainly in scope then.

> involves a lot of line re-writes just for the name change

There is no increase in the number of line rewrites. The PR introduces a new function. My function renaming request does not affect the number of line rewrites in the PR AFAICT.

> cleanup moving these functions callers into namespace Esi later would involve re-touching all those lines to remove the new Esi:: prefix

That reason may be valid although we do not require deleting unnecessary namespaces from otherwise unchanged code. This reason is also rather weak unless you expect to proceed with that cleanup very soon. Adding a `TODO: Move inside Esi` comment/reminder may be a good idea. And you can still drop the function name suffixes now. 

I do not insist on any of those changes.